integrate forensics-api for reference build search (#199)
jenkinsci/code-coverage-api-plugin: README.md

##### 3. Enable "Publish Coverage Report" publisher in the Post-build Actions.
##### 4. Add your coverage tool adapter and specify reports path.
![alt text](./images/config-add-adapter.png "Add coverage adapter")

> ##### 5. (Optional) Install the [forensics-api](https://github.com/jenkinsci/forensics-api-plugin) plugin 

Reviewer: this is a mandatory dependency of the plugin? So remove optional?

Author: No if you do not use the forensics-api and find the reference branch, there wont be a delta coverage calculation but thats ok.

Reviewer: This is a mandatory dependency:

![image](https://user-images.githubusercontent.com/21194782/119312362-2fd55f00-bc6a-11eb-916d-037ba7afab09.png)

That means it will be automatically installed for the user, and this step being worded wrong?

Author: The user has to trigger the reference search of the forensics api manually. If this is done, i am searching fir the reference build and calculate the diff. Otherwise the reference build wont be found and bo diff is calculated. I am not sure if if get your point?

Reviewer: The bit I am objecting to is '(Optional) Install the [forensics-api](https://github.com/jenkinsci/forensics-api-plugin) plugin'

The user cannot do this as it is force installed for them as it's a mandatory dependency

Author: Oh yes, of course.. will change this, thanks!

##### 3. Enable "Publish Coverage Report" publisher in the Post-build Actions.
##### 4. Add your coverage tool adapter and specify reports path.
![alt text](./images/config-add-adapter.png "Add coverage adapter")

> ##### 5. (Optional) Let discover the reference build to get delta coverage compared to the reference build. 
> Use the `discoverGitReferenceBuild()` of the [forensics-api](https://github.com/jenkinsci/forensics-api-plugin) plugin.
> You can use it directly and don't have to install the dependency, as it's a mandatory dependency!

Reviewer 2: It is a little bit more complex: the forensics API ist mandatory, but does not provide the `discoverGitReferenceBuild` step. This step is part of the git-forensics plugin (the only implementation of the API plugin so far). But using the forensics API  helps anyway since you can use it in freestyle jobs as well to compute a reference build. 

So I suggest to shorten as (if someone wants to read the details, then the documentation of the forensics API should be enough). 
```suggestion
##### 5. (Optional) Use the [forensics-api](https://github.com/jenkinsci/forensics-api-plugin) plugin to discover the reference build that is used to compute a delta report of the coverage results. 
```
